modules/service/src/main/scala/lucuma/odb/service/GuideService.scala

```diff
@@ -219,15 +220,18 @@ object GuideService {
         start:           Timestamp,
         end:             Timestamp,
         tracking:        ObjectTracking,
-        shapeConstraint: ShapeExpression
+        shapeConstraint: ShapeExpression,
+        wavelength:      Wavelength,
+        constraints:     ConstraintSet
       ): Either[Error, ADQLQuery] =
         (tracking.at(start.toInstant), tracking.at(end.toInstant))
           .mapN { (a, b) =>
+            val brightnessConstraints = gaiaBrightnessConstraints(constraints, GuideSpeed.Slow, wavelength)
             // Make a query based on two coordinates of the base of an asterism over a year
             CoordinatesRangeQueryByADQL(
               NonEmptyList.of(a.value, b.value),
               shapeConstraint,
-              ags.widestConstraints.some
+              brightnessConstraints.some
```

Comment:
the reason too use the widest constraiints is for caching. You'll get more stars but you'll be able to reuse the results for more stringent queries
As a plain function using widest will be less efficient but with cached results (like explore) it won't. I guess it depends on the context

Reply:
Makes sense. In the odb, we won't be caching the results from gaia, so I'm trying to reduce processing time by having fewer candidates.

Reply:
Should we have two entry points then?  so it is suitable for either case?

Reply:
I'm not sure I see the point. This can only be called from within the GuideService, so if we wanted to start caching, other changes would be required as well. Alternatively, `getGaiaQuery` could be called with `ags.widestConstraints`